model_compression_toolkit/qat/pytorch/quantizer/ste_rounding/uniform_ste.py

```diff
@@ -0,0 +1,117 @@
+        """
+        Initialize a TrainableWeightQuantizer object with parameters to use
+        for the quantization.
+
+        Args:
+            quantization_config: node quantization config class
+        """
+        super().__init__(quantization_config,
+                         qi.QuantizationTarget.Weights,
+                         [qi.QuantizationMethod.UNIFORM])
+        self.num_bits = self.quantization_config.weights_n_bits
+        self.min_int = 0
+        self.max_int = 2 ** self.num_bits - 1
+        self.max_values = quantization_config.weights_quantization_params[RANGE_MAX]
+        self.min_values = quantization_config.weights_quantization_params[RANGE_MIN]
+        self.min_max_shape = np.asarray(self.max_values).shape
+        self.max = np.reshape(self.max_values,
+                              [-1]) if self.quantization_config.weights_per_channel_threshold else float(
+            self.max_values)
+        self.min = np.reshape(self.min_values,
+                              [-1]) if self.quantization_config.weights_per_channel_threshold else float(
+            self.min_values)
+
+        self.quantizer_parameters = {}
+
+    def initialize_quantization(self,
+                                tensor_shape: torch.Size,
+                                name: str,
+                                layer: nn.Module) -> Dict[str, nn.Parameter]:
+        """
+        Add min and max variables to layer.
+        Args:
+            tensor_shape: Tensor shape the quantizer quantize.
+            name: Prefix of variables names.
+            layer: Layer to add the variables to. The variables are saved
+            in the layer's scope.
+
+        Returns:
+            Dictionary of new variables.
+        """
+        _shape = len(self.min) if self.quantization_config.weights_channels_axis else ()
+        _val = torch.ones(_shape) * self.min
+        self.fq_min = nn.Parameter(to_torch_tensor(_val), requires_grad=False)
+
+        _shape = len(self.max) if self.quantization_config.weights_channels_axis else ()
+        _val = torch.ones(_shape) * self.max
+        self.fq_max = nn.Parameter(to_torch_tensor(_val), requires_grad=False)
+
+        # save the quantizer added parameters for later calculations
+        self.quantizer_parameters = {FQ_MIN: self.fq_min, FQ_MAX: self.fq_max}
+
+        return self.quantizer_parameters
+
+    def forward(self,
+                 inputs: nn.Parameter,
+                 training: bool) -> nn.Parameter:
+        """
+        Quantize a tensor
+        Args:
+            inputs: Input tensor to quantize.
+            training: whether in training mode or not
+        Returns:
+            quantized tensor
+        """
+        max_range_tensor = self.max_values
+        min_range_tensor = self.min_values
+
+        # adjusts the quantization rage so the quantization grid include zero.
+        a, b = fix_range_to_include_zero(min_range_tensor, max_range_tensor, self.num_bits)
```

Comment:
Are you sure that this is needed here? why not also in the symmetric_ste?

Reply:
yes. this part is relevant only for uniform quantization